### ISSUE TITLE
Fix Request: False positive in AvoidForEachInStream

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#530]
example with only logging in forEach in stream:

```
   public static void logStackTrace(ServiceExceptionLogger.LogAction logAction, Throwable exception, int batchSize) {

        if (exception == null || exception.getStackTrace() == null || exception.getStackTrace().length == 0) {
            return;
        }
        // Our tool drops messages when the stacktrace is too long. Split into batches and log separate to circumvent.
        Arrays.stream(splitStackTraceIntoBatches(exception, batchSize)) // <- false positive
                .forEach(stackTraceElements ->
                        // log as warn - we can put the elements back together based on the traceId (MDC)
                        logAction.log(Arrays.stream(stackTraceElements).map(StackTraceElement::toString)
                                .collect(Collectors.joining(", " + System.lineSeparator()))));
    }

```
